controllers/operatorconfigmap_controller.go

```diff
@@ -320,7 +323,64 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{}, nil
 }
 
-func (c *ClusterVersionReconciler) createOrUpdate(obj client.Object, f controllerutil.MutateFn) error {
+func (c *OperatorConfigMapReconciler) deletionPhase() (reconcile.Result, error) {
+	ocsPvsPresent, err := c.hasOCSVolumes()
+	if err != nil {
+		c.log.Error(err, "unable to verify PVs presence prior deletion of ceph resources")
+		return ctrl.Result{}, err
+	}
+	if ocsPvsPresent {
+		c.log.Info("unable to delete ceph resources, PVs consuming client resources are present")
+		return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+	if err := csi.DeleteCSIDriver(c.ctx, c.Client, csi.GetCephFSDriverName()); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete cephfs CSIDriver")
+		return ctrl.Result{}, err
+	}
+	if err := csi.DeleteCSIDriver(c.ctx, c.Client, csi.GetRBDDriverName()); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete rbd CSIDriver")
+		return ctrl.Result{}, err
+	}
+	if err := c.Client.Delete(c.ctx, c.scc); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete SCC")
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+func (c *OperatorConfigMapReconciler) hasOCSVolumes() (bool, error) {
+	// get all the storage class
+	storageClassList := storagev1.StorageClassList{}
+	if err := c.Client.List(c.ctx, &storageClassList); err != nil {
+		return false, fmt.Errorf("unable to list storage classes: %v", err)
+	}
+
+	// create a set of storage class names who are using the client's provisioners
+	ocsStorageClass := make(map[string]bool)
+	for i := range storageClassList.Items {
+		storageClass := &storageClassList.Items[i]
+		if (storageClassList.Items[i].Provisioner == csi.GetCephFSDriverName()) || (storageClassList.Items[i].Provisioner == csi.GetRBDDriverName()) {
```

Comment:
I don't think in this statement you require extra pair of parens.
```suggestion
		if storageClassList.Items[i].Provisioner == csi.GetCephFSDriverName() || storageClassList.Items[i].Provisioner == csi.GetRBDDriverName() {
```